hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializerBuilder.java

```diff
@@ -134,6 +134,11 @@ else if ( creationState.isDynamicInstantiation() ) {
 			if ( cacheAccess != null ) {
 				// Do batch initialization instead of batch loading if the parent entity is cacheable
 				// to avoid putting entity state into the cache at a point when the association is not yet set
+				if ( entityPersister.getRepresentationStrategy()
```

Comment:
Should this rather be part of the first if condition in this method?

Reply:
not really, it fine if it is a BATCH_LOAD but probably we have to do this check also at line 108 and 123 where we may return BATCH_INITIALIZE.

Reply:
@beikov what do you think of the new change?

Reply:
LGTM